alexandria/core/models.py

```diff
@@ -171,38 +168,11 @@ def clone(self):
 
         self.pk = None
         self.save()
+        new_name = f"{self.pk}_{latest_original.name}"
 
-        storage = File.content.field.storage
         latest_original.pk = None
         latest_original.document = self
-        if isinstance(storage, S3Storage):
-            new_name = f"{self.pk}_{latest_original.name}"
-            copy_args = {
-                "CopySource": {
-                    "Bucket": settings.ALEXANDRIA_S3_BUCKET_NAME,
-                    "Key": latest_original.content.name,
-                },
-                # Destination settings
-                "Bucket": settings.ALEXANDRIA_S3_BUCKET_NAME,
-                "Key": new_name,
-            }
-
-            if settings.ALEXANDRIA_ENABLE_AT_REST_ENCRYPTION:
-                copy_args["CopySourceSSECustomerKey"] = storage.ssec_secret
-                copy_args["CopySourceSSECustomerAlgorithm"] = "AES256"
-                copy_args["SSECustomerKey"] = storage.ssec_secret
-                copy_args["SSECustomerAlgorithm"] = "AES256"
-
-            storage.bucket.meta.client.copy_object(**copy_args)
-            latest_original.content = new_name
-            latest_original.save()
-        else:
-            with NamedTemporaryFile() as tmp:
-                temp_file = Path(tmp.name)
-                with temp_file.open("w+b") as file:
-                    file.write(latest_original.content.file.file.read())
-                    latest_original.content = DjangoFile(file)
-                    latest_original.save()
+        latest_original.content.copy(new_name)
```

Comment:
second thought on saving the parent object within the field content: appears somewhat over the top and is probably quite unexpected behaviour

```suggestion
        latest_original.content.copy(new_name)
        latest_original.save()
```

Reply:
If it does not get saved a copy of the file will exist which doesnt belong to anywhere. I feel like that is worse behavior

Reply:
If you save it it's the same the other way round where the origial doesn't belong anywhere. Looks like there is not really a solution to avoid loose copies if copy is just called randomly :)

So we should at least point that out in a comment/docstring

Reply:
So fine with me to keep the saving within the field context

Reply:
Added a docstring